fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java

```diff
@@ -118,6 +121,38 @@ private Plan groupToTreeNode(Group group) {
         return result.withChildren(childrenNode);
     }
 
+    private static class GroupExpressionAdapter {
+        private final GroupExpression groupExpr;
+
+        public GroupExpressionAdapter(GroupExpression groupExpr) {
+            this.groupExpr = groupExpr;
+        }
+
+        public GroupExpression getGroupExpr() {
+            return groupExpr;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            GroupExpressionAdapter that = (GroupExpressionAdapter) o;
+            if (that.groupExpr.getPlan() instanceof LogicalOlapScan) {
```

Comment:
why special judgment for `OlapScan`?

Reply:
The class is a package class for GroupExpression because the equals() of the latter can not divide UnboundRelation from LogicalScanOlap as their logical properties are the same, but actually they are not the same.